spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/graphql/GraphQlAutoConfiguration.java

```diff
@@ -154,6 +169,25 @@ DataFetcherExceptionResolver annotatedControllerConfigurerDataFetcherExceptionRe
 		return annotatedControllerConfigurer.getExceptionResolver();
 	}
 
+	@ConditionalOnClass(ScrollPosition.class)
+	@Configuration(proxyBeanMethods = false)
+	static class GraphQlDataAutoConfiguration {
+
+		@Bean
+		@ConditionalOnMissingBean
+		CursorStrategy<ScrollPosition> cursorStrategy() {
+			return CursorStrategy.withEncoder(new ScrollPositionCursorStrategy(), CursorEncoder.base64());
+		}
+
+		@Bean
+		GraphQlSourceBuilderCustomizer cursorStrategyCustomizer(CursorStrategy<ScrollPosition> cursorStrategy) {
+			ConnectionFieldTypeVisitor connectionFieldTypeVisitor = ConnectionFieldTypeVisitor.create(
+					List.of(new WindowConnectionAdapter(cursorStrategy), new SliceConnectionAdapter(cursorStrategy)));
+			return (builder) -> builder.typeVisitors(List.of(connectionFieldTypeVisitor));
+		}
+
+	}
```

Comment:
If a `CursorStrategy` bean is declared by the application, it may not be parameterized by a different type or it may support both `ScrollPosition` in addition to something else. Maybe we could use this:

```java
@Bean
@ConditionalOnMissingBean
CursorStrategy<?> cursorStrategy() {
	return CursorStrategy.withEncoder(new ScrollPositionCursorStrategy(), CursorEncoder.base64());
}

@Bean
GraphQlSourceBuilderCustomizer cursorStrategyCustomizer(CursorStrategy<?> cursorStrategy) {
	if (cursorStrategy.supports(ScrollPosition.class)) {
			CursorStrategy<ScrollPosition> strategy = (CursorStrategy<ScrollPosition>) cursorStrategy;
			// ...
	}
}
```

Reply:
Good point! I've updated my PR accordingly.